models/utilities/default_model_mod.f90

```diff
@@ -283,6 +300,162 @@ subroutine pert_model_copies(state_ens_handle, ens_size, pert_amp, interf_provid
 
 end subroutine pert_model_copies
 
+!--------------------------------------------------------------------
+
+!> Reads in model_nml:model_state_variables and returns a
+!> state_var_type state_vars with nvars ; netcdf variable names ;
+!> qtys (kinds) ; clamp values ; updates
+!
+!> Verifies that the namelist was filled in correctly, and checks
+!> that there are valid entries for the dart_kind.
+
+subroutine get_state_variables_clamp(nml_state_vars, MAX_STATE_VARIABLES, use_clamping, state_vars)
```

Comment:
What's the thinking in have MAX_STATE_VARIABLES passed in vs. in default_model_mod?

Reply:
@hkershaw-brown I agree that MAX_STATE_VARIABLES should just be in default_model_mod. Why I ended up passing it in as an argument from the various model_mods is that the value for MAX_STATE_VARIABLES varies in each. For example, aether has it set to 100, some have it at 20, but most have it set to 10. Therefore, I thought that we might want to allow the various model_mods to use different values for MAX_STATE_VARIABLES as opposed to having it as a parameter in default_model_mod. 

So I guess my main question to do this instead is - what should we have be the value for MAX_STATE_VARIABLES in default_model_mod? It seems inefficient to use 100, but then I'm not sure if this will be suitable for aether and other future models that might use more than 10 or 20

Reply:
Inefficient in terms of memory?  Are you worried about stack size?

(vtablenamelength = 64) * (5 columns) * (100 rows) = 31.74KB 
I would not worry about this. 

For computational efficiency, 

* you exit the loop early, so you only loop over the actual number of state variables
```
! Loop through the variables array to get the actual count of the number of variables
do ivar = 1, MAX_STATE_VARIABLES
   ! If the first element in the row is an empty string, the loop has exceeded the extent of the variables
   if (nml_state_vars(5*ivar-4) == '') then
      state_vars%nvars = ivar-1
      exit
   endif
enddo
```

* in get_state_variables_x
```
subroutine get_state_variables_clamp(nml_state_vars, MAX_STATE_VARIABLES, use_clamping, state_vars)

character(len=*),   intent(inout) :: nml_state_vars(:) <- this here you're better off being explicit about the (size)
                                                                                             also intent(out). The compiler can make better 
                                                                                            optimizations. But again tiny array, called once. Not that big 
                                                                                            a deal
integer,               intent(in) :: MAX_STATE_VARIABLES
logical,               intent(in) :: use_clamping
type(state_var_type), intent(out) :: state_vars
```